torrent.go

```diff
@@ -1,13 +1,110 @@
 package tpb
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+)
+
 // Torrent represents a Torrent
 type Torrent struct {
-	Name        string
-	Leechers    int
-	Seeders     int
-	User        string
-	Magnet      string
-	Size        uint64
-	Category    TorrentCategory
-	SubCategory TorrentCategory
+	ID          int             `json:"id"`
+	Category    TorrentCategory `json:"category"`
+	Status      UserStatus      `json:"status"`
+	Name        string          `json:"name"`
+	NumFiles    int             `json:"num_files"`
+	Size        uint64          `json:"size"`
+	Seeders     int             `json:"seeders"`
+	Leechers    int             `json:"leechers"`
+	User        string          `json:"username"`
+	Added       time.Time       `json:"added"`
+	Description string          `json:"descripiton"`
+	InfoHash    string          `json:"info_hash"`
+	ImdbID      string          `json:"imdb"`
+	// Magnet      string          `json:"magnet"`
+}
+
+// GetMagnet returns a Magnet for a torrent
+func (t *Torrent) GetMagnet() string {
+	var trackers = "tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce&tr=udp%3A%2F%2F9.rarbg.to%3A2920%2Fannounce&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=udp%3A%2F%2Ftracker.internetwarriors.net%3A1337%2Fannounce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.pirateparty.gr%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.cyberia.is%3A6969%2Fannounce"
+	return fmt.Sprintf("magnet:?xt=urn:btih:%s&dn=%s&%s", t.InfoHash, t.Name, trackers)
+}
+
+// UnmarshalJSON is a custom unmarshal function to handle timestamps and
+// boolean as int and convert them to the right type.
+func (t *Torrent) UnmarshalJSON(data []byte) error {
+	dataBytes := bytes.NewReader(data)
+	var aux struct {
+		ID          flexInt    `json:"id"`
+		Category    flexInt    `json:"category"`
+		Status      string     `json:"status"`
+		Name        string     `json:"name"`
+		NumFiles    flexInt    `json:"num_files"`
+		InfoHash    string     `json:"info_hash"`
+		Description string     `json:"descr"`
+		Leechers    flexInt    `json:"leechers"`
+		Seeders     flexInt    `json:"seeders"`
+		User        string     `json:"username"`
+		Size        flexInt    `json:"size"`
+		Added       flexInt    `json:"added"`
+		ImdbID      flexString `json:"imdb"`
+	}
+
+	// Decode json into the aux struct
+	if err := json.NewDecoder(dataBytes).Decode(&aux); err != nil {
```

Comment:
Do we need a decoder ? We have `[]byte`, maybe we could just use:
```go
json.Unmarshal(data, &aux)
```

Reply:
True!